src/core/EdFi.DataManagementService.Core/Backend/UpdateCascadeHandler.cs

```diff
@@ -0,0 +1,263 @@
+            // find the specific element(s) in the list that needs to be updated.
+            // For example, we need to turn this: "$.classPeriods[*].classPeriodReference.classPeriodName"
+            // into this: $.classPeriods[?(@.classPeriodReference.classPeriodName == "Third period edited" && @.classPeriodReference.schoolId == 123)]
+
+            List<string> filters = [];
+            var firstSegment = "";
+
+            foreach (
+                var originalIdentityJsonPath in originalIdentityJsonPaths
+                    .Where(x => x != null)
+                    .Select(x => JsonPath.Parse(x!.GetValue<string>()))
+            )
+            {
+                var referenceJsonPath =
+                    referencingReferenceJsonPaths.Single(a =>
+                        a != null
+                        && a.SelectNodeFromPathAs<string>("$.identityJsonPath", _logger)
+                            == originalIdentityJsonPath.ToString()
+                    )
+                    ?? throw new InvalidOperationException(
+                        $"Unexpected null finding {referencingResourceName}.documentPathsMapping.{originalDocumentResourceName}.identityJsonPath = {originalIdentityJsonPath}"
+                    );
+
+                var referenceJsonPathString =
+                    referenceJsonPath.SelectNodeFromPathAs<string>("$.referenceJsonPath", _logger)
+                    ?? string.Empty;
+
+                // split at the array marker
+                var split = referenceJsonPathString.Split("[*]");
+                if (firstSegment == string.Empty)
+                {
+                    firstSegment = split[0];
+                }
+                referenceJsonPathString = split[1];
+
+                originalIdentityJsonPath
+                    .Evaluate(originalEdFiDoc.AsNode())
+                    .Matches.TryGetSingleValue(out JsonNode? originalValue);
+                if (originalValue == null)
+                {
+                    throw new InvalidOperationException(
+                        $"Unexpected error finding original value for {originalIdentityJsonPath}"
+                    );
+                }
+
+                switch (originalValue.GetValueKind())
+                {
+                    case JsonValueKind.Number:
+                        filters.Add($"@{referenceJsonPathString} == {originalValue.GetValue<long>()}");
+                        break;
+                    default:
+                        filters.Add($"@{referenceJsonPathString} == \"{originalValue.GetValue<string>()}\"");
+                        break;
+                }
+            }
+
+            var filterExpression = $"{firstSegment}[?({string.Join(" && ", filters)})]";
+
+            // Evaluate
+            var filterPath = JsonPath.Parse(filterExpression);
+            var pathResult = filterPath.Evaluate(returnEdFiDoc);
+            // should have at least one match
```

Comment:
Is it also invalid if there is more than one match?

Reply:
I think more than one match is surprising, but the right thing to do is just update them all. Maybe log a warning that doc with this uuid has duplicates?